### PR TITLE
[Infrastructure] Avoid downloading browsers during source build

### PR DESCRIPTION
**Pull request by javiercn**
Setup PUPPETEER_SKIP_DOWNLOAD=1 to avoid downloading browsers as part of source build

### CODE REVIEW

File: eng/common/templates/steps/source-build.yml [hunk 1]
```diff
@@ -83,6 +83,10 @@ steps:
       assetManifestFileName=SourceBuild_${{ parameters.platform.name }}.xml
     fi
 
+    # Setup environment variables to support offline builds before calling into
+    # the build script.
+    export PUPPETEER_SKIP_DOWNLOAD=1
```
Comment (MichaelSimons) [inline eng/common/templates/steps/source-build.yml:88]: This is not a robust solution - it doesn't get applied to the product source-build or when a dev builds aspnetcore in SB mode locally.  This should be set in the [SourceBuild.props](https://github.com/dotnet/aspnetcore/blob/main/eng/SourceBuild.props).  e.g. specify in `EnvironmentVariables=` in the npm build `exec`

Comment (javiercn) [reply eng/common/templates/steps/source-build.yml:88]: Fair enough, I'll update it

File: eng/Npm.Workspace.nodeproj [hunk 2]
```diff
@@ -22,7 +22,11 @@
 
   <Target Name="Restore" DependsOnTargets="_VerifyNPMCache">
     <Message Text="Restoring NPM packages..." Importance="high" />
-    <Exec Command="npm ci --offline" WorkingDirectory="$(MSBuildThisFileDirectory).." />
+    <PropertyGroup>
```
Comment (wtgodbe) [inline eng/Npm.Workspace.nodeproj:25]: Don't we need this in `SourceBuild.props` instead? https://github.com/dotnet/aspnetcore/blob/b61a93689c4ac61b571904c0e0e665914083c71a/eng/SourceBuild.props#L110-L112

Comment (javiercn) [reply eng/Npm.Workspace.nodeproj:25]: Yes, you are right

**Review by javiercn — COMMENTED** [review]
Update in SourceBuild.props

**Comment by MichaelSimons** [comment]
I opened a draft PR in the VMR with a modification to run a full CI build in order to validate these changes - https://github.com/dotnet/dotnet/pull/67

**Review by wtgodbe — APPROVED** [review]

**Comment by mthalman** [comment]
> I opened a draft PR in the VMR with a modification to run a full CI build in order to validate these changes - [dotnet/dotnet#67](https://github.com/dotnet/dotnet/pull/67)

Looks like that's working (the two failing jobs there are expected).

**Review by mthalman — APPROVED** [review]

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**